ikarus/utils/dirichletvalues.hh

```diff
@@ -82,6 +82,33 @@ public:
       Dune::Functions::forEachBoundaryDOF(basis_, lambda);
     }
   }
+  /**
+   * \brief Function to fix (set boolean values to true or false) degrees of freedom on the boundary using a
+   * SubSpaceBasis of the Basis used to create the DirichletValues
+   *
+    This function takes a callback function, which will be called with the boolean vector of fixed boundary
+   * degrees of freedom and the usual arguments of `Dune::Functions::forEachBoundaryDOF`.
+   *
+   * \param f A callback function
+   * \param ssb A subspace basis (compile-time-checked to be an actual subspace basis of basis)
+   */
+  template <typename F, typename SSB>
+  requires(std::is_same_v<typename std::remove_cvref_t<SSB>::RootBasis, Basis>)
+  void fixBoundaryDOFs(F&& f, SSB&& ssb) {
```

Comment:
This function is not needed.

Can you instead please refactor the old function, such that

```cpp
template <typename F,typename  TreePath= Dune::TypeTree::hybridTreePath>
  void fixBoundaryDOFs(F&& f,TreePath&& treePath={}) {
//...

Dune::Functions::forEachBoundaryDOF(subSpaceBasis(basis_,std::forward<TreePath>(treePath)), lambda);
}
```

Reply:
Additionally you use a external subBasis, which may not be the same as the one constructed from `basis_`